zap/src/main/java/org/zaproxy/zap/extension/ascan/CustomScanDialog.java

```diff
@@ -643,6 +643,8 @@ private JPanel getTechPanel() {
                             javax.swing.border.EtchedBorder.RAISED));
 
             techPanel.add(scrollPane, LayoutHelper.getGBC(0, 0, 1, 1, 1.0D, 1.0D));
+        } else {
+            getTechTree().refresh();
```

Comment:
This resets the tree state, losing the previous selections.

Reply:
Any idea on how I can refresh the tree including the state when a tech gets added? I always got errors in awt stuff,and I'm not so into awt.

Reply:
Reworked and working now.